app/frontend/src/features/profile/hooks/referencesHooks.ts

```diff
@@ -118,6 +120,9 @@ export function useWriteFriendReference(userId: number) {
     {
       onSuccess: () => {
         queryClient.invalidateQueries([availableWriteReferencesKey(userId)]);
+        queryClient.invalidateQueries([
+          referencesReceivedKey(userId, ReferenceType.REFERENCE_TYPE_FRIEND),
+        ]);
```

Comment:
```suggestion
        queryClient.invalidateQueries(referencesReceivedBaseKey);
```
Same as above once we made the small change in `src/queryKeys.ts`